File: 2020-04-cit-why-golang/07-entities/analyst.java
class Analyst {
    protected String name;
    protected String email;

    String getDivision() {
        return "dev-team";
    }

    String getContact() {
        return this.name + " - " + this.email + " - " + this.getDivision();
    }
}

class Manager extends Analyst {
    private String division;

    @Override
    String getDivision() {
        return this.division;
    }

    Manager(String division, String name, String email) {
        this.name = name;
        this.email = email;
        this.division = division;
    }
}

class MainClass {
    public static void main(String... args) {

        Manager m = new Manager("management", "Hugo", "h@g.com");
        System.out.println(m.getContact());

    }
}

File: 2020-04-cit-why-golang/07-entities/analyst.go
package main

import "fmt"

type Analyst struct {
	Name  string
	Email string
}

func (a Analyst) Division() string {
	return "dev-team"
}

func (a Analyst) Contact() string {
	return a.Name + " - " + a.Email + " - " + a.Division()
}

type Manager struct {
	Analyst
	MDivision string
}

func (m Manager) Division() string {
	return m.MDivision
}

func main() {

	m := Manager{
		Analyst: Analyst{
			Name:  "Hugo",
			Email: "h@g.com",
		},
		MDivision: "Management",
	}

	fmt.Println(m.Contact())
}
